crates/sui-core/src/unit_tests/move_package_upgrade_tests.rs

```diff
@@ -1114,3 +1130,108 @@ async fn test_different_versions_across_calls() {
 
     assert!(effects.status.is_ok(), "{:#?}", effects.status);
 }
+
+#[tokio::test]
+async fn test_conflicting_versions_across_calls() {
+    // publishes base package at version 1
+    let mut runner = UpgradeStateRunner::new("move_upgrade/base").await;
+
+    // publish a depender package at version 1 that depends on the base package at version 1
+    let (_, module_bytes, dep_ids) = build_upgrade_test_modules_with_dep_addr(
+        "dep_on_upgrading_package_upgradeable",
+        [
+            ("base_addr", runner.package.0),
+            ("dep_on_upgrading_package", ObjectID::ZERO),
+        ],
+        [("package_upgrade_base", runner.package.0)],
+    );
+    let (depender_package, depender_cap) = runner.publish(module_bytes, dep_ids).await;
+
+    // publish base package at version 2
+    let pt1 = build_upgrade_txn(
+        runner.package.0,
+        "stage1_basic_compatibility_valid",
+        runner.upgrade_cap,
+    );
+
+    let TransactionEffects::V1(effects) = runner.run(pt1).await;
+    assert!(effects.status.is_ok(), "{:#?}", effects.status);
+
+    let base_v2_package = effects
+        .created
+        .iter()
+        .find(|(_, owner)| matches!(owner, Owner::Immutable))
+        .unwrap()
+        .0;
+
+    // publish a depender package at version 2 that depends on the base package at version 2
+    let pt2 = {
+        let mut builder = ProgrammableTransactionBuilder::new();
+        let current_package_id = depender_package.0;
+        // Now recompile the depending package with the upgraded dependency
+        // Currently doesn't work -- need to wait for linkage table to be added to the loader.
```

Comment:
nit: comment out of date (I'm guessing this is copypasta from another comment that's also out-of-date?)

Reply:
Copypasta indeed and the other test does work now of course so removed it in both cases.